docs/Microservices/RPC/troubleshootingRPC.md

```diff
@@ -16,4 +33,25 @@ The table below details possible errors that may occur when activating or de-act
 | Error: amt password DOES NOT match stored version for Device 6c4243ba-334d-11ea-94b5-caba2a773d00 | Ensure you have provided the `--password` flag for the `--cmd/-c` you are trying to execute, and that it is the password you used when provisioning the device. |
 | Unable to connect to websocket server. Please check url. | After ensuring you can reach your server. Ensure that the certificate common name on the server matches the FQDN/IP of your host address. |
 | Error while activating the AMT in admin mode. | Check the logs on the RPS server. | 
-| The rpc.exe fails to connect. | If a device has already been provisioned, [unprovision](../../Topics/MEBX/unprovision.md) it and then reprovision. To deactivate and reactivate devices, see the Mircoservices section for RPC, [Command Examples](commandsRPC.md) | 
+| The rpc.exe fails to connect. | If a device has already been provisioned, [unprovision](../../Topics/MEBX/unprovision.md) it and then reprovision. To deactivate and reactivate devices, see the Mircoservices section for RPC, [RPC Activate/Deactivate Examples](commandsRPC.md) | 
+
+# UI Toolkit
+- If you encounter an error during the installation, verify the prerequisites and version numbers, such as Node.js* LTS, by consulting the tutorial [Add MPS UI Toolkit Controls to a WebUI](../../Tutorials/uitoolkit.md). 
+- If adding a control results in an error, double-check the device ID, mpsServer IP address value, and authToken.
+
+# General Troubleshooting Tips
+
+If a configuration becomes unworkable, it may be necessary clean up the environment by:
+
+- Deactivating the managed device
+- Unprovisioning the managed device
+- Stopping all Docker services
+- Disabling the management engine
+
+Do all the above if it becomes necessary to reset your environment completely. See instructions below.
+
+1. **Deactivate the Managed Device:** Use rpc.exe as described in [RPC Activate/Deactivate Examples](commandsRPC.md#RPCexamples)
+2. **Unprovision the Managed Device:** See [Unprovisioning](../../Topics/MEBX/unprovision.md)
```

Comment:
@matt-primrose Is there a difference between an RPC Deactivation and an unprovision via MEBX? Don't they accomplish the same thing? Or does MEBX reset certain other values?

Reply:
They essentially accomplish the same thing.  Via MEBX, there is an option to perform either a Full or Partial Unprovision.  Full Unprovision is the same as what RPC does.  Partial unconfigures AMT, but then turns on remote configuration where AMT will reach out to a configuration server directly to get re-provisioned.  We don't support the Partial Unprovision flow currently.